posts/GSoC 2019.md

```diff
@@ -0,0 +1,64 @@
+
+## What is PyCanChemAndYouCan2(ChemCheck)
+
+PyCanChemAndYouCan2 is a web-based debugging tool developed for Cantera to visualize and diagnose syntax errors and chemical errors in input files. Users can upload  input files and visualize their errors on the website, and they can also download it after editing the input files.
+
+## Why Would I Need It
+
+Cantera uses input files containing mechanism of reactions because mechanism information is the fundamental to do thermo-kinetic calculation. The input files are required to be in .cti format, and users are able to gain them from various sources (e.g.: writing .cti file by hands, adopted from various sources and repositories on the web, or via a hybrid of these two approaches etc.).   However, .cti files always include errors since they are from different sources, which could cause  inaccurate results. The errors include syntax errors and chemical errors, and it is not likely to be found manually especially in large mechanism files. In this case, a debugging tool helping users to find and fix errors is very useful. 
+## What Errors Can Be Visualized or Fixed:
+The errors being detected by the application will include but not limited:
+Syntax errors
+
+-   duplicate lines
+-   missing digits from end of line
+-   missing 'E'
+-   whatever else we find
+
+Chemistry errors
+
+-   discontinuous polynomicals
+-   exceeding collision limit
+-   inconsistent thermo for explicit reverse rates
+-   negative rate coefficient (as a  [result](https://github.com/comocheng/wiki/issues/375#)  of sum of arrhenius or PLOG expressions, eg.  [here](https://github.com/Cantera/cantera-website/issues/77))
+-   excess stiffness
+-   making  [error](https://github.com/comocheng/wiki/issues/375#)  messages from CVODES easier to understand
+-   making errors from CVODES easier to catch (raise a python Error)
+-   dead-end pathways
+## How Can I Achieve It
+
+The debugging tool PyCanChemAndYouCan2 is a web-based application developed in Django 2.2.  Users can upload original files and download revised files from the website, and the application contains  ACE code editor to allow users edit their files on the website. For Debugging script, Cantera has `ck2cti.py` file to convert different file format, so the debugging part will be test different defective mechanism files and update `ck2cti.py` file. The ideal condition is that the application can help users find the errors and sugget the fixes or fix errors automatically. However, it might be a long-term goal, so we make some minimum goal for this GSoC project.
+Here are the basic ideas about the project:
+ -   a script that takes in an input file and returns an input file with added comments, which describe detected errors.
+-   a web site that makes it easy to interact with that script (which runs on the web server), and renders the comments in an attractive way
+
+The idea is that the script can be used, developed, tested, as a stand-alone thing without the website.  
```

Comment:
This paragraph seems like it is copy-pasted from another comment. Can you make it into your own words?